woocommerce/woocommerce-ios
# [Deposit Summary] Payment menu analytics for deposit summary

Part of: #11234 


## Description


This PR adds two of the five analytics for the new Deposit Summary view on the Payments menu.

The others will be implemented in the Deposit Summary view itself, so make sense to do in a separate PR as this is getting large.

The reason for the excessive size (for two analytics) is that we did not previously support fake/copy for WooFoundation files – this PR rectifies that, and also fixes enum fakes which previously did not declare the module they were from, leading to potential conflicts. Going commit-by-commit _might_ be an easier way to review this.

### Known issues
The `Shown` event was being tracked when the HubMenu was loaded, because we fetch the deposit summary on init. This is prevented with a flag, but I'm intending to move that to an `onAppear` on the depositSummary itself. Since we don't currently have a viewModel for that, I'm leaving it as is for the moment.

The `Error` event is tracked for stores which don't have WooPayments installed. I'll handle this separately, as I may deal with it by only fetching data for stores with WooPayments, or by suppressing the `restNoRoute` tracking. If other extensions implement the `overview-all` API, it would be nice to show their information here, but that does seem unlikely as it's not a published api spec.

## Testing instructions


1. Launch the app
2. Switch to a WooPayments Store
3. Navigate to `Menu > Payments`
4. Observe that when the deposit summary shows, the `PAYMENTS_HUB_DEPOSIT_SUMMARY_SHOWN` event is tracked, with the number of currencies.

### Errors

1. Repeat the above test, with Proxyman configured to breakpoint on the `/wc/v3/payments/deposits/overview-all` endpoint
2. Edit the response to a 500 error, then continue
3. Observe that the `PAYMENTS_HUB_DEPOSIT_SUMMARY_ERROR` event is tracked, along with the error description.

## Screenshots

![Screenshot showing the two above events tracked in the Xcode console](https://github.com/woocommerce/woocommerce-ios/assets/2472348/49c12c62-710c-4586-8c5f-09f014c8015b)


---
